operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -697,18 +747,115 @@ private void addIncludesSearchParameters(String compartmentMemberType, Multivalu
+                List<Entry> allEntries, List<String> resourceIds, 
+                FHIRResourceHelpers resourceHelper, List<String> extraResources) throws Exception {
+        if (extraResources == null || extraResources.isEmpty()) {
+            return;
+        }
+        String url = FHIRRequestContext.get().getOriginalRequestUri().substring(0, FHIRRequestContext.get().getOriginalRequestUri().indexOf("/Patient/"));
+        List<Resource> resourceList = new ArrayList<Resource>();
+        for (Entry entry: newEntries) {
+            // Look up entries by reference
+            List<Reference> references = ReferenceFinder.getReferences(entry.getResource());
+            for (Reference reference: references) {
+                ReferenceValue referenceValue = ReferenceUtil.createReferenceValueFrom(reference, compartmentMemberType);
+                if (referenceValue.getType().equals(ReferenceValue.ReferenceType.LITERAL_RELATIVE)) {
+                    String externalId = reference.getReference().as(com.ibm.fhir.model.type.String.class).getValue();
+                    String[] splitString = externalId.split("/");
+                    String type = splitString[0];
+                    if (!resourceIds.contains(externalId) && extraResources.contains(type) && (type.equals(ResourceType.Value.LOCATION.value()) || type.equals(ResourceType.Value.MEDICATION.value())
+                            || type.equals(ResourceType.Value.ORGANIZATION.value()) || type.equals(ResourceType.Value.PRACTITIONER.value()))) {
+                        resourceList.add(resourceHelper.doRead(type, splitString[1], false, false, null, null).getResource());
+                        resourceIds.add(externalId);
+                    }
+                } else if (referenceValue.getType().equals(ReferenceValue.ReferenceType.LITERAL_ABSOLUTE)) {
+                    String externalUrl = reference.getReference().as(com.ibm.fhir.model.type.String.class).getValue();
+                    int lastIndexOf = externalUrl.lastIndexOf("/");
+                    String externalId = externalUrl.substring(lastIndexOf+1, externalUrl.length());
+                    String tempString = externalUrl.substring(0,lastIndexOf);
+                    lastIndexOf = tempString.lastIndexOf("/");
+                    String type = tempString.substring(lastIndexOf+1, tempString.length());
+                    if (!resourceIds.contains(externalId) && extraResources.contains(type) && (type.equals(ResourceType.Value.LOCATION.value()) || type.equals(ResourceType.Value.MEDICATION.value())
+                            || type.equals(ResourceType.Value.ORGANIZATION.value()) || type.equals(ResourceType.Value.PRACTITIONER.value()))) {
+                        resourceList.add(resourceHelper.doRead(type, externalId, false, false, null, null).getResource());
+                        resourceIds.add(type + "/" + externalId);
+                    }
```

Comment:
I don't think we should try to do anything with absolute URLs here
```suggestion
```

Reply:
Code removed.